### PR TITLE
Fix: doc versioning by header

### PR DESCRIPTION
**Pull request by kperreau**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ???

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: [1495](https://github.com/nestjs/swagger/issues/1495)

## What is the new behavior?
Versioning documentation is now working for header versioning.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
     "@nestjs/platform-fastify": "9.0.11",
     "@types/jest": "28.1.7",
     "@types/js-yaml": "4.0.5",
-    "@types/lodash": "4.14.182",
+    "@types/lodash": "4.14.184",
```
Comment (micalevisk) [inline package.json:43]: why do we need to upgrade this? :thinking:

Comment (kperreau) [reply package.json:43]: It's already updated on master, it's a fail from my rebase: https://github.com/nestjs/swagger/commit/67c78e4bb1abf2a4c882dd040c4e53ae28534053

Comment (micalevisk) [reply package.json:43]: coud you rebase it again? looks like we got some conflicts to fix

Comment (kperreau) [reply package.json]: Sure, it's done

**Comment by kamilmysliwiec** [comment]
Apologies for not reviewing this sooner!

Could you please resolve conflicts? Also, any chance we can add an e2e test too (with enabled versioning by header)?

**Comment by david-induro** [comment]
Any updates on this? Looking forward to being able to use it! :)

**Comment by jointhejourney** [comment]
is there anything remaining to merge this? happy to contribute to this PR to get it merged

**Comment by kamilmysliwiec** [comment]
We're still waiting for e2e tests to be added

**Comment by dalaverd** [comment]
Any news on this?

**Pull request closed without merging**